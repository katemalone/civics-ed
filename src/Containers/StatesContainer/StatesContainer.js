import React from 'react'; 
import { connect } from 'react-redux';
import StatesCard from '../StatesCard/StatesCard';
import { addStatesList } from '../../Actions'
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';


export const StatesContainer = ({ statesList }) => {

  let statesInfo = statesList.map(state => {

    return (
      <StatesCard 
        key={state.abbreviation.toLowerCase()}
        name={state.name} 
        id={state.abbreviation.toLowerCase()}
        statesImg={state.abbreviation}
        /> 
    )
  })
  
  return(
    <section className="StatesContainer"> 
      {statesInfo}
    </section>
  )
}

export const mapStateToProps = ({ statesList }) => ({
  statesList
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addStatesList
  },dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(StatesContainer)

StatesContainer.propTypes = {
  addStateList: PropTypes.func,
  statesList: PropTypes.array,
}