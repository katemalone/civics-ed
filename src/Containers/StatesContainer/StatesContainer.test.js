import React from 'react';
import { shallow } from 'enzyme';
import { StatesContainer } from './StatesContainer';

describe('StatesContainer', () => {

  it('should match snapShot', () => {
    const mockStatesList = [
      {
        "name": "Alabama",
        "abbreviation": "AL"
      },
      {
        "name": "Alaska",
        "abbreviation": "AK"
      }
    ]
    const wrapper = shallow(
      <StatesContainer statesList={mockStatesList} />);
    expect(wrapper).toMatchSnapshot()
  });
});