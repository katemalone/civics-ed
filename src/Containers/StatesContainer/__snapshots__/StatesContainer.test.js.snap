// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatesContainer should match snapShot 1`] = `
<section
  className="StatesContainer"
>
  <Connect(StatesCard)
    id="al"
    key="al"
    name="Alabama"
    statesImg="AL"
  />
  <Connect(StatesCard)
    id="ak"
    key="ak"
    name="Alaska"
    statesImg="AK"
  />
</section>
`;
